cmd/influx_inspect/verify_seriesfile/verify_seriesfile.go

```diff
@@ -0,0 +1,298 @@
+	if err := fs.Parse(args); err != nil {
+		return err
+	}
+
+	return cmd.run(*dataDir, *dbPath, *filePath)
+}
+
+// run performs all of the validations, preferring a filePath, then a dbPath,
+// and then a dataDir.
+func (cmd *Command) run(dataDir, dbPath, filePath string) error {
+	if filePath != "" {
+		_, err := cmd.runFile(filePath)
+		return err
+	}
+	if dbPath != "" {
+		_, err := cmd.runFile(filepath.Join(dbPath, "_series"))
+		return err
+	}
+
+	dbs, err := ioutil.ReadDir(filepath.Join(dataDir, "data"))
+	if err != nil {
+		return err
+	}
+
+	for _, db := range dbs {
+		if !db.IsDir() {
+			continue
+		}
+		_, err := cmd.runFile(filepath.Join(dataDir, "data", db.Name(), "_series"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// runFile performs validations on a series file. The error is only returned if
+// there was some fatal problem with operating, not if there was a problem with
+// the series file.
+func (cmd *Command) runFile(filePath string) (valid bool, err error) {
+	logger := logger.New(cmd.Stderr).With(zap.String("path", filePath))
+	logger.Info("starting validation")
+
+	defer func() {
+		if rec := recover(); rec != nil {
+			logger.Error("panic validating file",
+				zap.String("recovered", fmt.Sprint(rec)))
+			valid = false
+		}
+	}()
+
+	partitionInfos, err := ioutil.ReadDir(filePath)
+	if os.IsNotExist(err) {
+		logger.Error("series file does not exist")
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	// check every partition
+	for _, partitionInfo := range partitionInfos {
+		pLogger := logger.With(zap.String("partition", partitionInfo.Name()))
+		pLogger.Info("validating partition")
+		partitionPath := filepath.Join(filePath, partitionInfo.Name())
+
+		segmentInfos, err := ioutil.ReadDir(partitionPath)
+		if err != nil {
+			return false, err
+		}
+		segments := make([]*tsdb.SeriesSegment, 0, len(segmentInfos))
+		ids := make(map[uint64]idData)
+
+		// check every segment
+		for _, segmentInfo := range segmentInfos {
+			segmentPath := filepath.Join(partitionPath, segmentInfo.Name())
+
+			segmentID, err := tsdb.ParseSeriesSegmentFilename(segmentInfo.Name())
+			if err != nil {
+				continue
+			}
+
+			sLogger := pLogger.With(zap.String("segment", segmentInfo.Name()))
+			sLogger.Info("validating segment")
+
+			segment := tsdb.NewSeriesSegment(segmentID, segmentPath)
+			if err := segment.Open(); err != nil {
+				sLogger.Error("opening segment", zap.Error(err))
+				return false, nil
+			}
+			defer segment.Close()
+
+			if offset, err := cmd.validateSegment(segment, ids); err != nil {
+				sLogger.Error("iterating over segment",
+					zap.Int64("offset", offset),
+					zap.Error(err))
+				return false, nil
+			}
+
+			segments = append(segments, segment)
+		}
+
+		// validate the index if it exists
+		indexPath := filepath.Join(partitionPath, "index")
+		_, err = os.Stat(indexPath)
+		if os.IsNotExist(err) {
+			continue
+		}
+		if err != nil {
+			return false, err
+		}
```

Comment:
This is mostly personal preference but I generally group my error checks in a single `if`/`else if` block:

```go
if _, err := os.Stat(indexPath); os.IsNotExist(err) {
	continue
} else if err != nil {
	return false, err
}
```